src/main/java/com/github/manolo8/darkbot/core/objects/facades/AstralGateProxy.java

```diff
@@ -89,7 +90,11 @@ public void update() {
             this.upgradeLevel = API.readInt(itemData + 0x2C);
 
             this.lootId = API.readString(itemData, 0x48);
-            this.status = API.readString(itemData, 0x60);
+
+            itemStatsArr.update(API.readMemoryLong(address + 0x38));
+            synchronized (UPDATE_LOCKER) {
```

Comment:
is this going to be used in the ui thread? if not you probably do not need synchronization.

Maybe we should synchronize at the facade manager level to avoid having it all around

Reply:
It is true that the stats are not necessary because they will not be updated with UI changes.